src/graphcast_agent/mod.rs

```diff
@@ -47,6 +49,119 @@ pub const REGISTRY_SUBGRAPH: &str =
 /// A constant defining the goerli network subgraph endpoint.
 pub const NETWORK_SUBGRAPH: &str = "https://gateway.testnet.thegraph.com/network";
 
+#[derive(Debug, thiserror::Error)]
+pub enum ConfigError {
+    #[error("Validate the input: {0}")]
+    ValidateInput(String),
+    #[error("Unknown error: {0}")]
+    Other(anyhow::Error),
+}
+
+#[derive(Clone)]
+pub struct GraphcastAgentConfig {
+    pub wallet_key: String,
+    pub radio_name: &'static str,
+    pub registry_subgraph: String,
+    pub network_subgraph: String,
+    pub graph_node_endpoint: String,
+    pub boot_node_addresses: Vec<Multiaddr>,
+    pub graphcast_namespace: Option<String>,
+    pub subtopics: Vec<String>,
+    pub waku_node_key: Option<String>,
+    pub waku_host: Option<String>,
+    pub waku_port: Option<String>,
+    pub waku_addr: Option<String>,
+}
+
+impl GraphcastAgentConfig {
+    #[allow(clippy::too_many_arguments)]
+    pub fn new(
+        wallet_key: String,
+        radio_name: &'static str,
+        registry_subgraph: String,
+        network_subgraph: String,
+        graph_node_endpoint: String,
+        boot_node_addresses: Option<Vec<String>>,
+        graphcast_namespace: Option<String>,
+        subtopics: Option<Vec<String>>,
+        waku_node_key: Option<String>,
+        waku_host: Option<String>,
+        waku_port: Option<String>,
+        waku_addr: Option<String>,
+    ) -> Result<Self, GraphcastAgentError> {
+        let boot_node_addresses = convert_to_multiaddrs(&boot_node_addresses.unwrap_or(vec![]))
+            .map_err(|_| GraphcastAgentError::ConvertMultiaddrError)?;
+
+        Ok(GraphcastAgentConfig {
```

Comment:
I think `validate_set_up` should be called before this returns, and then it doesn't need to be called when creating a GraphcastAgent at https://github.com/graphops/graphcast-sdk/pull/200/files#diff-53dc15c3ce98e4c8261a55e83bad27c23dc644642ce782ac30ff6e4b6ed59748R248